In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import scipy.stats as stat
import warnings
warnings.filterwarnings('ignore')

In [2]:
df1=pd.read_csv('C:/Users/HP/Downloads/Week_8_Q&A_dataset - Sheet1.csv')

In [3]:
df1.head()

Gender Test preparation  Total Marks
0    male             none           14
1  female             none           28
2  female             none           18
3  female             none           48
4  female             none           21

In [4]:
df1.shape

(28, 3)

In [5]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 28 entries, 0 to 27
Data columns (total 3 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   Gender            28 non-null     object
 1   Test preparation  28 non-null     object
 2   Total Marks       28 non-null     int64 
dtypes: int64(1), object(2)
memory usage: 800.0+ bytes


# 1. Descriptive Analysis

### Count

In [6]:
# gender
df1.value_counts('Gender')

Gender
female    16
male      12
dtype: int64

In [7]:
# test completion
df1.value_counts('Test preparation')

Test preparation
none         21
completed     7
dtype: int64

### Measures of Central Tendancy

In [8]:
# mean mark
df1['Total Marks'].mean()

32.32142857142857

In [9]:
# mean mark based on gender
df1.groupby('Gender')['Total Marks'].mean()

Gender
female    33.750000
male      30.416667
Name: Total Marks, dtype: float64

In [10]:
# mean mark based on the test completion
df1.groupby('Test preparation')['Total Marks'].mean()

Test preparation
completed    44.571429
none         28.238095
Name: Total Marks, dtype: float64

Clearly, on an average, those students who completed the test preparation has scored much higher than those who didn't complete the course

### Measures of Dispersion

In [11]:
# variance
df1['Total Marks'].var()

155.11507936507937

In [12]:
# variance based on gender
df1.groupby('Gender')['Total Marks'].var()

Gender
female    165.533333
male      148.083333
Name: Total Marks, dtype: float64

In [13]:
#variance based on test preparartion
df1.groupby('Test preparation')['Total Marks'].var()

Test preparation
completed     11.285714
none         135.990476
Name: Total Marks, dtype: float64

 The marks scored by the students who completed the test preparation has much smaller compared to the others, therefore their marks are more consistent.

In [14]:
# range
max_mark=df1['Total Marks'].max()
min_mark=df1['Total Marks'].min()
range_mark=max_mark-min_mark
print('maximum mark scored is: ',max_mark,' and minimum mark is: ',min_mark)
print('the range of total marks is: ',range_mark)

maximum mark scored is:  50  and minimum mark is:  12
the range of total marks is:  38


### Measures of Position

In [15]:
q1,q2,q3=np.quantile(df1['Total Marks'],(.25,.5,.75))
print('25% of the students scored mark below ',q1)
print('Top 25% of the students scored above ',q3)

25% of the students scored mark below  22.5
Top 25% of the students scored above  43.25


# 2. Checking equality of scores for male and female students

H0: The marks scored by male And female students are equal, i.e, H0: mu1 Equal to mu2 

H1: The marks scored by male And female students are significantly different, i.e, H1: mu1 Notequal to mu2

sig.level Is 0.05

In [16]:
df_male=df1[(df1['Gender']=='male')] #data of male students

In [17]:
df_female=df1[(df1['Gender']=='female')] #data of female students

In [18]:
stat.ttest_ind(df_male['Total Marks'],df_female['Total Marks'],alternative='two-sided')

Ttest_indResult(statistic=-0.6940885268848668, pvalue=0.4937822317960763)

The pvalue=0.4937822317960763 is greater than significance level=.05

Hence we cannot reject the null hypothesis.

The marks scored by male and female students didnot differ significantly

# 3. Effect of test preparaion

 H0: Marks, scored by the students who completed test preparation and students who didn't complete, are equal
 i.e, H0: mu1=mu2

 H1: Students who completed test preparation, scored more marks than the others.
 i.e, H1: mu1>mu2

 sig.level=.05

In [19]:
df_completed=df1[(df1['Test preparation']=='completed')]

In [20]:
df_uncomplete=df1[(df1['Test preparation']=='none')]

In [21]:
stat.ttest_ind(df_completed['Total Marks'],df_uncomplete['Total Marks'],alternative='greater')

Ttest_indResult(statistic=3.6143637527769217, pvalue=0.0006335646228660053)

 The pvalue 0.0006335646228660053<.05
 
 Hence we reject our null hypothesis.
 
 The test preparation was helpful to students